src/app/app-component.service.ts

```diff
@@ -0,0 +1,187 @@
+import { Injectable } from '@angular/core';
+import { ListWeatherPoints, WeatherData, WidgetListWeatherPoints, WidgetWeatherData } from './interfaces';
+
+const PRESSURE_PASCAL_TO_MERCURY_BAROMETER = 1/1.333;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AppComponentService {
+
+  // This method changes the input data to the format required by the weather display component
+  // It distributes the disordered information to the days of the period (period 5 days)
+  weatherDataTransformation(inputData: WeatherData): WidgetWeatherData[] {
+    let outputData: WidgetWeatherData[] = new Array();
+
+    const limitDays = 5; //limit of days to show
+    let firstDay: Date = new Date(inputData.list[0].dt_txt);
+
+    function timeDiff (firstDate: Date, secondDate: Date): number {
+      firstDate.setHours(0,0,0,0);
+      secondDate.setHours(0,0,0,0);
+      return (+secondDate - +firstDate)/(1000*3600*24);
+    };
+
+    // Filtering out unnecessary timestamps
+    let listWeatherPoints: ListWeatherPoints[] = inputData.list.filter((element) => {
+      return timeDiff(firstDay,(new Date(element.dt_txt))) < limitDays;
+    });
```

Comment:
JFI: Could be simplified and listWeatherPoints is not changed so could be used const instead of let(will be fixed with eslint later I think)
```
const listWeatherPoints: ListWeatherPoints[] = inputData.list.filter((element) => 
    timeDiff(firstDay,(new Date(element.dt_txt))) < limitDays
);
```